packages/time-picker/src/TimePicker/utils/generateRegexFromInput.ts

```diff
@@ -0,0 +1,24 @@
+import { DateSegment } from "@react-stately/datepicker"
+
+export const generateRegexFromInput = (segments: DateSegment[]): RegExp => {
```

Comment:
Why do we need this function? Do we actually need to generate a regex? Perhaps we could encapsulate the Regex inside, and return a function that just returns `true` or `false`. For now it could keep the regex and use its `.test` method — but encapsulate that.